File: Jack.Model/Model/Status.vb
<Serializable()> _
Public Class Status

#Region "Construtor"

    Public Sub New()
        Codigo = 0
        Descricao = String.Empty
    End Sub

    Public Sub New(strDescricao As String)
        Codigo = 0
        Descricao = strDescricao
    End Sub

    Public Sub New(intCodigo As Integer, strDescricao As String)
        Codigo = intCodigo
        Descricao = strDescricao
    End Sub


    Public Sub New(intCodigo As Integer, strDescricao As String, strPermiteSacola As String, strNivelStatus As String)
        Codigo = intCodigo
        Descricao = strDescricao
        PermiteSacola = strPermiteSacola
        NivelStatus = strNivelStatus
    End Sub

#End Region

    Public Overridable Property Codigo As Integer
    Public Overridable Property Descricao As String
    Public Overridable Property PermiteSacola As String
    Public Overridable Property NivelStatus As String

    Public ReadOnly Property PermiteSacolaDesc As String
        Get
            If PermiteSacola = "S" Then
                Return "Sim"
            Else
                Return "Não"
            End If
        End Get
    End Property
    Public ReadOnly Property NivelStatusDesc As String
        Get
            If NivelStatus = "F" Then
                Return "Família"
            ElseIf NivelStatus = "C" Then
                Return "Criança"
            Else
                Return "Todos"
            End If
        End Get
    End Property

End Class

File: Jack.Model/Model/Status.cs
using Jack.Library.Extensions;
using System;
using System.Collections.Generic;

namespace Jack.Model
{
    [Serializable()]
    public class Status : BaseModel<Status>
    {
 
        #region "Construtor"

        public Status() :base()
        {
            descricao = string.Empty;
        }

        public Status(string strDescricao) :this()
        {
            descricao = strDescricao;
        }

        public Status(int intCodigo, string strDescricao) : this()
        {
            Codigo = intCodigo;
            descricao = strDescricao;
        }

        public Status(int intCodigo, string strDescricao, string strPermiteSacola, string strNivelStatus) : this()
        {
            Codigo = intCodigo;
            descricao = strDescricao;
            permiteSacola = strPermiteSacola;
            nivelStatus = strNivelStatus;
        }

        #endregion

        #region "Fields"
        private string descricao;
        private string permiteSacola;
        private string nivelStatus;
        private string permiteSacolaDesc;
        private string nivelStatusDesc;
        private IList<Familia> familias;
        private IList<Criancas> criancas;
        #endregion

        public virtual string Descricao
        {
            get
            {
                return descricao;
            }
            set
            {
                descricao = value;
            }
        }

        public virtual string PermiteSacola
        {
            get
            {
                return permiteSacola;
            }
            set
            {
                permiteSacola = value;
                permiteSacolaDesc = permiteSacola.ToSimNao();
            }
        }

        public virtual string NivelStatus
        {
            get
            {
                return nivelStatus;
            }
            set
            {
                nivelStatus = value;
                nivelStatusDesc.ToNivel();
            }
        }


        public virtual string PermiteSacolaDesc
        {
            get
            {
                return permiteSacolaDesc;
            }
        }

        public virtual string NivelStatusDesc
        {
            get
            {
                return nivelStatusDesc;
            }
        }

        public virtual IList<Familia> Familias
        {
            get
            {
                return familias;
            }
            set
            {
                familias = value;
            }
        }

        public virtual IList<Criancas> Criancas
        {
            get
            {
                return criancas;
            }
            set
            {
                criancas = value;
            }
        }

    }
}
